onnxscript/rewriter/pattern.py

```diff
@@ -1566,6 +1567,12 @@ def _apply_to_graph_or_function(
                 if delta is None or tracer is not None:
                     continue
                 assert isinstance(delta, ReplacementSubgraph)
+                if delta.new_initializers and isinstance(graph_or_function, ir.Function):
+                    # TODO(rama): Can't add initializers to functions. But currently this is not
+                    # an issue, as we apply inlining before applying rewrite rules.
+                    continue
+                for initializer in delta.new_initializers:
+                    graph_or_function.initializers[initializer.name] = initializer
```

Comment:
Should we warn about overwriting initializers?

Reply:
Good question. I guess you mean if there is already an initializer with that name? This is also related to naming authority issues ... we could rename them also, but, for now, may be failing would be fine.

Reply:
I think there are instances when we want to keep/reuse the original name. Maybe a warning is fine for now? I don't have a strong opinion